pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -901,7 +903,22 @@ func validateBootOrder(field *k8sfield.Path, spec *v1.VirtualMachineInstanceSpec
 			}
 			bootOrderMap[order] = true
 		}
+
+		diskNames[disk.Name] = nil
 	}
+
+	// Validate that volumes match disks correctly
+	for idx, volume := range spec.Volumes {
+		if _, machingDiskExists := diskNames[volume.Name]; !machingDiskExists {
```

Comment:
@iholder-redhat volumes can also match `spec.Domain.Devices.Filesystems`

Reply:
see [1] for example.
[1] https://github.com/kubevirt/kubevirt/blob/master/pkg/virt-launcher/virtwrap/converter/converter.go#L1373

Reply:
Thanks!
`diskNames` is now `diskAndFilesystemNames` and contain both fs and disk names :)